spec/support/shared_examples/bulk_updatable_type.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+GENERATORS = {
+  string: ->(index) { "wat_#{index}" },
+  boolean: ->(index) { index.even? },
+  hstore: ->(index) { { "foo_#{index}" => "bar_#{index}" } },
+  datetime: ->(index) { Time.zone.now.in_time_zone('UTC').round - index.days },
+  jsonb: ->(index) { { "bee_#{index}" => "bizzle_#{index}" } },
+}.freeze
+
+RSpec.shared_examples 'bulk updatable type' do |type|
+  let(:getter_name) { "#{type}_value" }
+  let(:setter_name) { "#{type}_value=" }
+  let(:type) { type }
+
+  def model_values(models)
+    BulkUpdatableModel.where(id: models.map(&:id)).pluck(getter_name)
+  end
+
+  def generate_values(models, seed: 1)
+    models.each_with_index.map do |model, index|
+      value = GENERATORS.fetch(type).(index * seed)
```

Comment:
Not sure I understand `fetch(type).(index * seed)` what this is doing.  Maybe it's the `.` that is confusing me.

Reply:
This is a shorthand syntax. In ruby, `.()` is syntax sugar for `.call()`, so if something has a `.call()` method you can just use `.()`. We've got this linted so that in `JunkDrawer` we always use the shorthand, and in `Synchroform` we always use the long form. I'm open to re-evaluating that, but stuck with the style of the repo for this.